source/common/network/connection_impl.cc

```diff
@@ -518,7 +529,9 @@ void ConnectionImpl::updateWriteBufferStats(uint64_t num_written, uint64_t new_s
 ClientConnectionImpl::ClientConnectionImpl(Event::DispatcherImpl& dispatcher,
                                            Address::InstanceConstSharedPtr address)
     : ConnectionImpl(dispatcher, address->socket(Address::SocketType::Stream), address,
-                     getNullLocalAddress(*address), false) {}
+                     getNullLocalAddress(*address), false) {
+  markConnecting();
```

Comment:
nit: IMO it would be better to just pass a boolean to ConnectionImpl constructor saying whether to mark connecting or not vs. having a method.